WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderWebView.java

```diff
@@ -188,6 +207,34 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 return false;
             }
         }
+
+        @Override
+        public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+            // Intercept requests and add the WP.com authorization header
+            if (!TextUtils.isEmpty(mToken) && isImageUrl(url)) {
```

Comment:
It seems inefficient to do this for every image when most posts aren't private. How about we add a `hasPrivateImages` property to ReaderWebView that determines whether this is necessary? ReaderPostDetailFragment would set that property [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java#L721) based on `mPost.isPrivate`.